modules/java-modules/elastic-v2/src/main/java/org/syslog_ng/elasticsearch_v2/ElasticSearchOptions.java

```diff
@@ -35,212 +35,228 @@
+    public static String CONFIG_FILE = "resource";
+    public static String CONCURRENT_REQUESTS = "concurrent_requests";
+    public static String CLUSTER_URL = "cluster_url";
+    public static String FLUSH_TIMEOUT = "flush_timeout";
+
+    public static String SERVER_DEFAULT = "localhost";
+    private static final String PORT_DEFAULT = "0";
+    public static String MESSAGE_TEMPLATE_DEFAULT = "$(format-json --scope rfc5424 --exclude DATE --key ISODATE)";
+    public static String FLUSH_LIMIT_DEFAULT = "5000";
+    public static String CLIENT_MODE_TRANSPORT = "transport";
+    public static String CLIENT_MODE_NODE = "node";
+    public static String CLIENT_MODE_SHIELD = "shield";
+    public static String CLIENT_MODE_SEARCHGUARD = "searchguard";
+    public static String CLIENT_MODE_HTTP = "http";
+    public static String CLIENT_MODE_HTTPS = "https";
+    public static String SKIP_CLUSTER_HEALTH_CHECK = "skip_cluster_health_check";
+    public static String SKIP_CLUSTER_HEALTH_CHECK_DEFAULT = "false";
+    public static String JAVA_KEYSTORE_FILEPATH = "java_keystore_filepath";
+    public static String JAVA_KEYSTORE_PASSWORD = "java_keystore_password";
+    public static String JAVA_TRUSTSTORE_FILEPATH = "java_truststore_filepath";
+    public static String JAVA_TRUSTSTORE_PASSWORD = "java_truststore_password";
+    public static String HTTP_AUTH_TYPE = "http_auth_type";
+    public static String HTTP_AUTH_TYPE_NONE = "none";
+    public static String HTTP_AUTH_TYPE_BASIC = "basic";
+    public static String HTTP_AUTH_TYPE_CLIENTCERT = "clientcert";
+    public static String HTTP_AUTH_TYPE_BASIC_USERNAME = "http_auth_type_basic_username";
+    public static String HTTP_AUTH_TYPE_BASIC_PASSWORD = "http_auth_type_basic_password";
+    public static HashSet<String> CLIENT_MODES = new HashSet<String>(Arrays.asList(CLIENT_MODE_TRANSPORT,
+            CLIENT_MODE_NODE, CLIENT_MODE_SHIELD, CLIENT_MODE_HTTP, CLIENT_MODE_HTTPS, CLIENT_MODE_SEARCHGUARD));
+    public static HashSet<String> HTTP_AUTH_TYPES = new HashSet<String>(
+            Arrays.asList(HTTP_AUTH_TYPE_NONE, HTTP_AUTH_TYPE_BASIC, HTTP_AUTH_TYPE_CLIENTCERT));
+    public static final HashMap<String, Integer> DEFAULT_PORTS_BY_MODE = new HashMap<String, Integer>() {
+        {
+            this.put(CLIENT_MODE_HTTP, 9200);
+            this.put(CLIENT_MODE_HTTPS, 9200);
+            this.put(CLIENT_MODE_NODE, 9300);
+            this.put(CLIENT_MODE_TRANSPORT, 9300);
+            this.put(CLIENT_MODE_SEARCHGUARD, 9300);
         }
+    };
 
-        public boolean getSkipClusterHealthCheck() {
-                return options.get(SKIP_CLUSTER_HEALTH_CHECK).getValueAsBoolean();
-        }
+    public static String CLIENT_MODE_DEFAULT = CLIENT_MODE_TRANSPORT;
+    public static String HTTP_AUTH_TYPE_DEFAULT = HTTP_AUTH_TYPE_NONE;
+    public static String CONCURRENT_REQUESTS_DEFAULT = "1";
+    public static String FLUSH_TIMEOUT_DEFAULT = "1";
```

Comment:
Hmm somehow github shows my comment outdated earlier. Trying again....
Shouldn't this one be integer?

Reply:
No, here is everything is string and it is converted later.